server/tls_config.go

```diff
@@ -0,0 +1,55 @@
+package server
+
+import (
+	"crypto/tls"
+	fmt "fmt"
+	"strings"
+
+	"github.com/prometheus/exporter-toolkit/web"
+)
+
+// Collect all cipher suite names and IDs recognized by Go, including insecure ones.
+func allCiphers() map[string]web.Cipher {
+	acceptedCiphers := make(map[string]web.Cipher)
+	for _, suite := range tls.CipherSuites() {
+		acceptedCiphers[suite.Name] = web.Cipher(suite.ID)
+	}
+	for _, suite := range tls.InsecureCipherSuites() {
+		acceptedCiphers[suite.Name] = web.Cipher(suite.ID)
+	}
+	return acceptedCiphers
+}
+
+func stringToCipherSuites(s string) ([]web.Cipher, error) {
+	if s == "" {
+		return nil, nil
+	}
+	ciphersSlice := []web.Cipher{}
+	possibleCiphers := allCiphers()
+	for _, cipher := range strings.Split(s, ",") {
+		intValue, ok := possibleCiphers[cipher]
+		if !ok {
+			return nil, fmt.Errorf("Cipher suite %q not recognized", cipher)
```

Comment:
[Nit] Conventionally Go errors are not capitalized, right?
```suggestion
			return nil, fmt.Errorf("cipher suite %q not recognized", cipher)
```

Reply:
You can get points for reporting / fixing the same thing upstream!
https://github.com/kubernetes/kubernetes/blob/d5f39ebe4d2700eccf094c54b36c5c46302fb4ec/staging/src/k8s.io/component-base/cli/flag/ciphersuites_flag.go#L107

Reply:
Good point @bboreham :D Taking a look there.